model/events/synchronization.go

```diff
@@ -1,16 +1,39 @@
 package events
 
 import (
+	"github.com/onflow/flow-go/model"
 	"github.com/onflow/flow-go/model/cluster"
 	"github.com/onflow/flow-go/model/flow"
 )
 
+// SyncedBlock is an internal message passed from the chainsync engine to the consensus message hub.
+// It contains a block obtained through the chainsync protocol.
+// These blocks have not been validated and should be considered untrusted inputs,
+// however they are guaranteed to be structurally valid: StructureValid()==nil
 type SyncedBlock struct {
 	OriginID flow.Identifier
 	Block    *flow.Block
 }
 
+var _ model.StructureValidator = (*SyncedBlock)(nil)
+
+// StructureValid checks basic structural validity of the message and ensures no required fields are nil.
+func (m *SyncedBlock) StructureValid() error {
```

Comment:
when I read 
> checks basic structural validity of the message
I feel it is a reasonable assumption to think "I can assume that [`ProposerSigData`](https://github.com/onflow/flow-go/blob/04b4bcaf206cac75e956a557a959668b100d9356/model/flow/header.go#L45) has the appropriate length, otherwise the block is clearly structurally invalid". I think documentation really needs to be really explicitly here about what is checked in detail.